recipes/libsolace/all/conanfile.py

```diff
@@ -0,0 +1,62 @@
+import os
+
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+class LibsolaceConan(ConanFile):
+    name = "libsolace"
+    description = "High performance components for mission critical applications"
+    topics = ("HPC", "High reliability", "P10", "solace", "performance", "c++", "conan")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/abbyssoul/libsolace"
+    license = "Apache-2.0"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = "shared=False", "fPIC=True"
+    generators = "cmake"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _supported_cppstd(self):
+        return ["17", "gnu17", "20", "gnu20"]
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os == "Windows":
+          raise ConanInvalidConfiguration("This library is not yet compatible with Windows")
+        if self.settings.compiler.cppstd and not self.settings.compiler.cppstd in self._supported_cppstd:
+          raise ConanInvalidConfiguration("This library requires c++17 standard or higher. {} required".format(self.settings.compiler.cppstd))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self, parallel=True)
+        cmake.definitions["BUILD_TESTING"] = "OFF"
```

Comment:
```suggestion
```
According CMake, this definition is not being used.

Reply:
done